pkg/build/api/v1beta3/conversion.go

```diff
@@ -160,10 +164,12 @@ func init() {
 		},
 
 		func(in *newer.BuildParameters, out *BuildConfigSpec, s conversion.Scope) error {
-			err := s.DefaultConvert(&in.Strategy, &out.Strategy, conversion.IgnoreMissingFields)
-			if err != nil {
+			if err := s.Convert(&in.Strategy, &out.Strategy, 0); err != nil {
```

Comment:
The correct conversion.  DefaultConvert silently ignores mismatches - this verifies mismatches don't happen.